packages/vetrina/src/Vetrina/Vetrina.purs

```diff
@@ -121,34 +126,41 @@ initialState =
   , products: []
   , productSelection: Nothing
   , paymentMethod: CreditCard
+  , unexpectedError: mempty
   }
 
 didMount :: Self -> Effect Unit
-didMount self = do
-  sentryDsn <- sentryDsn_
-  logger <- Sentry.mkLogger sentryDsn Nothing
-  self.setState _ { logger = logger }
-  -- Before rendering the form, we need to:
-  -- 1. fetch the user if access token is found in the browser
-  Aff.launchAff_ do
-    Aff.finally
-      -- When user has been fetched, hide loading spinner
-      (liftEffect $ self.setState \s -> s { isLoading = Nothing })
-      do
-        -- Try to login with local storage information and set user to state
-        tryMagicLogin self
-
-        products <- liftEffect $ case self.props.products of
-          Right p -> pure p
-          Left err -> do
-            self.setState _ { purchaseState = PurchaseFailed $ UnexpectedError ""  }
-            logger.error err
-            throwError err
-
-        liftEffect $ self.setState _ { products = products }
-        -- If there is only one product given, automatically select that for the customer
-        when (length products == 1) $
-          liftEffect $ self.setState _ { productSelection = head products }
+didMount self =
+  case self.props.unexpectedError of
+    Just view ->
+      self.setState _ { purchaseState   = UnexpectedServerError
+                      , unexpectedError = view
+                      }
+    Nothing -> do
+      sentryDsn <- sentryDsn_
+      logger <- Sentry.mkLogger sentryDsn Nothing
+      self.setState _ { logger = logger }
+      -- Before rendering the form, we need to:
+      -- 1. fetch the user if access token is found in the browser
+      Aff.launchAff_ do
+        Aff.finally
+          -- When user has been fetched, hide loading spinner
+          (liftEffect $ self.setState \s -> s { isLoading = Nothing })
+          do
+            -- Try to login with local storage information and set user to state
+            tryMagicLogin self
+
+            products <- liftEffect $ case self.props.products of
+              Right p -> pure p
+              Left err -> do
+                self.setState _ { purchaseState = PurchaseFailed $ UnexpectedError ""  }
```

Comment:
For instance this would not render anything, although we cannot render vetrina successfully, as the products were broken.

Reply:
I mean, it would need to render the `UnexpectedServerError` view